src/apps/lemmafinance/optimism/lemmafinance.LemmaSynth.token-fetcher.ts

```diff
@@ -0,0 +1,121 @@
+      '0xd1a988b024C55d7bAaBB07fD531d63A4E19e3B4C', // LemmaPerp
+      '0xA7C657a94Eb9571f4e94F49943Af1130e6D7337c', // LemmaAAVE
+    ];
+
+    const perpLemmas = [
+      '0x29b159aE784Accfa7Fb9c7ba1De272bad75f5674', // PerpLemmaWeth
+      '0xe161C6c9F2fC74AC97300e6f00648284d83cBd19', // PerpLemmaWbtc
+      '0xdd4d71D3563C24E38525661896e1d01Fd8c2c9A5', // PerpLemmaLink
+      '0x119f85ECFcFBC1d7033d266192626202Df7dbDf2', // PerpLemmaCRV
+      '0x13c214b430fE304C4C6437F3564A690cd4e4f23B', // PerpLemmaPerp
+      '0xFE1EB36d31ead771Fd5E051ee8CC424dB6416567', // PerpLemmaAAVE
+    ];
+
+    const imageURL = 'src/apps/lemmafinance/assets/';
+    const tokenImages = [
+      'LETH.png',
+      'LBTC.png',
+      'LLink.png',
+      'LCRV.png',
+      'LPERP.png',
+      'LAAVE.png'
+    ];
+
+    const multicall = this.appToolkit.getMulticall(network);
+    const tokens = await Promise.all(
+      LemmaSynths.map(async (lemmaSynth, i) => {
+        // Instantiate a smart contract instance pointing to the jar token address
+        const contract = this.lemmafinanceContractFactory.lemmaSynth({
+          address: lemmaSynth,
+          network,
+        });
+
+        const perpLemmaContract = this.lemmafinanceContractFactory.perpLemma({
+          address: perpLemmas[i],
+          network,
+        });
+
+        // Request the symbol, decimals, ands supply for the jar token
+        const [name, symbol, decimals, supplyRaw, collateral, indexPrice] = await Promise.all([
+          multicall.wrap(contract).name(),
+          multicall.wrap(contract).symbol(),
+          multicall.wrap(contract).decimals(),
+          multicall.wrap(contract).totalSupply(),
+          multicall.wrap(contract).tailCollateral(),
+          multicall.wrap(perpLemmaContract).getIndexPrice(),
+        ]);
+
+        // Denormalize the supply
+        const supply = Number(supplyRaw) / 10 ** decimals;
+        const tokens: any = [collateral];
+        const price = Number(indexPrice) / 10 ** decimals;
+        const pricePerShare = Number(1);
+
+        // // As a label, we'll use the underlying label (i.e.: 'LOOKS' or 'UNI-V2 LOOKS / ETH'), and suffix it with 'Jar'
+        const label = `${name} (${symbol})`;
+        // // For images, we'll use the underlying token images as well
+        const images = [imageURL + tokenImages[i]];
+        // // For the secondary label, we'll use the price of the jar token
+        const secondaryLabel = buildDollarDisplayItem(price);
+        // // And for a tertiary label, we'll use the APY
+        // const tertiaryLabel = `${(apy * 100).toFixed(3)}% APY`;
+
+        // Create the token object
+        const token: AppTokenPosition = {
+          type: ContractType.APP_TOKEN,
+          appId,
+          groupId,
+          address: lemmaSynth,
+          network,
+          symbol,
+          decimals,
+          supply,
+          tokens,
+          price,
+          pricePerShare,
+          dataProps: {},
```

Comment:
You will also want to add liquidity within the dataProps

Reply:
Yes Thanks for that, I added liquidity please take a look now below commit.